src/interactions/coulomb.jl

```diff
@@ -21,48 +21,46 @@ Coulomb(nl_only=false) = Coulomb(
     138.935458 / 70.0 # Treat ϵr as 70 for now
 )
 
-@inline @inbounds function force!(forces,
-                                    inter::Coulomb{C},
-                                    s::Simulation,
-                                    i::Integer,
-                                    j::Integer) where C
-    i == j && return
-    dr = vector(s.coords[i], s.coords[j], s.box_size)
+@inline @inbounds function force(inter::Coulomb{C},
+                                    coord_i,
+                                    coord_j,
+                                    atom_i,
+                                    atom_j,
+                                    box_size) where C
+    coord_i == coord_j && return zero(coord_i)
```

Comment:
This causes a performance drop as we now compare floating point numbers instead of integers.

Reply:
Okay I'll probably just remove it then, and make a note to add a `SelfInteraction` type so that a couple of the examples keep working.

Reply:
The problem with removing it would be in the out of place case, where we must take care what we sum.

Reply:
True, I was thinking of passing a diagonal array of ones as an extra broadcasting argument. Then a wrapper function returns `zero()` if it sees a one, representing an i-i interaction, otherwise it calculates the force as normal.